src/data/part.lean

```diff
@@ -7,6 +7,38 @@ import data.equiv.basic
 
 /-!
 # Partial values of a type
+
+What to do when you only want to provide a term of `α` against the proof of your favorite
+proposition? Use partial values™!
+"But we already have `option α`..." I hear. Que nenni! What if your proposition isn't decidable
+because you don't believe in the excluded middle? Use partial values™!
+Ask for partial values at your local mathlib® distribution. To be found under reference `part α`.
```

Comment:
I'm afraid this is a bit too colloquial, and doesn't blend well with the rest of mathlib.
(Sorry to be that guy...)

I understand there isn't too much to say about `part`, but I think it might make more sense to have a (slightly expanded?, e.g. mention an application/example) version of the docstring that currently sits on `part`.

Reply:
Don't worry, I wasn't expecting this to make it to mathlib 😄